### DISCUSSION TITLE
Extract statistics on used spaces / radii / shadows / borders / etc?

### DISCUSSION THREAD

**Discussion by sergeylukin** [system-ui/theme-ui#1849] [Q&A]
Hi,

I'm considering adding a space in the middle of my theme's spaces array (while my base space is 4 there is a jump between 16 and 32 which I initially thought would be OK however now I ended up needing 24 and 28), however in order to not break code that depends on >16 space token, I'd like to extract statistics on where it's used in order to refactor the code accordingly. Same would apply to any other theme tokens so my question is, how can I print a list of components (file/line number) that are using theme tokens like spaces / radii / shadows / etc?

Thanks

**Comment by hasparus** [comment]
I'd love to have a CLI for this, but right now I can offer a temporary solution.

Import `scales` from `@theme-ui/css`. It holds the mapping from style property into theme scale name, so we can use it to build a regex to search for all occurences of `[property]: ourOldValue`.

Alternatively, I’d try to hook into our `jsx` function (defined in`@theme-ui/css`).
